### DISCUSSION TITLE
how to sovle the flicker when display pic

### DISCUSSION THREAD

**Discussion by Kingwulin** [Bodmer/TFT_eSPI#2250] [Q&A - General]
Hi
I use a 240 * 320 screen
I using the code below to display a dynamic background and some small pictures
```
void loop()
{
  delay(1000);
  tft.pushImage(0,0,320,240,backgroup001);
  showMyFonts(10,10,"汉字测试",TFT_GREENYELLOW);
  tft.pushImage(70, 180, 45, 45, img_apple);
  delay(1000);
  tft.pushImage(0,0,320,240,backgroup002);
  showMyFonts(20,20,"汉字测试",TFT_BLUE);
  tft.pushImage(70, 180, 45, 45, img_apple);
  delay(1000);
  tft.pushImage(0,0,320,240,backgroup003);
  tft.pushImage(70, 180, 45, 45, img_apple);
  showMyFonts(40,50,"汉字测试",TFT_BLACK);
}
```
but there will be obvious flicker when displaying small pictures(below video)
I think it is a problem that the background and small pictures are displayed separately
So whether it is possible to write the data to the display cache first, and then call the display interface
Or whether there are other better solutions

https://user-images.githubusercontent.com/16952072/207220232-b45930a0-754f-4fee-820f-cbe19e2da05e.mp4

**Comment by Bodmer** [comment]
Only update the areas of the screen that change. Use sprites for complex areas. A full screen sprite will require 150Kbytes of RAM, so the maximum size you can use is dependant on the processor. It appears you are trying to create a game, so have a look at the approach taken here:
https://hackaday.io/project/164205-esp-little-game-engine

**Reply by Kingwulin** [reply]
I am not create a game,I want to display some simple emoticon，digits and  some different background pictures(3 day and 3 night) to form a simple animation
The following are the materials and the total effect
[UI.zip](https://github.com/Bodmer/TFT_eSPI/files/10217405/UI.zip)
